rf.c

```diff
@@ -71,7 +73,12 @@ unsigned int rf_get_node_id(void) {
 	unsigned int id = 1;
 	read(REG_NODEID_L, (unsigned char *) &id, sizeof(id));
 	return id;
-}	
+}
+
+void rf_set_node_id(unsigned int id) {
+		write(REG_NODEID_L, (unsigned char) id&0xFF);
+		write(REG_NODEID_H, (unsigned char) (id>>8));
```

Comment:
Did you try to change the "write" function to be like the read:

write(REG_NODEID_L, &id, sizeof(id));

I don't remember if the rf uC firmware handle consecutive write, but if it does for read, it should for write.